allennlp/interpret/attackers/input_reduction.py

```diff
@@ -0,0 +1,145 @@
+from copy import deepcopy
+from typing import List, Tuple
+import numpy as np
+import torch
+from allennlp.interpret.attackers.attacker import Attacker
+from allennlp.interpret.attackers import utils
+from allennlp.common.util import JsonDict, sanitize
+from allennlp.data import Instance
+from allennlp.data.fields import ListField, TextField, LabelField
+
+@Attacker.register('input-reduction')
+class InputReduction(Attacker):
+    """
+    Runs the input reduction method from `Pathologies of Neural Models Make Interpretations
+    Difficult` https://arxiv.org/abs/1804.07781, which removes as many words as possible
+    from the input without changing the model's prediction.
+    """
+    def attack_from_json(self, inputs: JsonDict = None,
+                         input_field_to_attack: str = 'tokens',
+                         grad_input_field: str = 'grad_input_1',
+                         ignore_tokens: List[str] = ["@@NULL@@"]):
```

Comment:
You're using `["@@NULL@@"]` as a default value to this function.  This is what pylint is complaining about.  It's not good practice in python, except in a few very rare cases.  See, e.g., [here](http://satran.in/2012/01/12/python-dangerous-default-value-as-argument.html).  Instead of this, use `None` as the default value, and inside the method body do `ignore_tokens = ["@@NULL@@"] if ignore_tokens is None else ignore_tokens`.  You could also just do `ignore_tokens = ignore_tokens or ["@@NULL@@"]`, except in this case that would make it so you are unable to specify `ignore_tokens=[]`, which is probably not what you want.